conf/SimpleConfiguration.php

```diff
@@ -0,0 +1,268 @@
+	{
+		if(is_null($config)) {
+			foreach($this->dynamicConfigs as $config) {
+				$val = "";
+				$expandedConfig = self::getVariableByAlias($config);
+				foreach($expandedConfig as $portion) {
+					if(is_array($portion)) {
+						$val .= $this->parseDynamicConfigs($portion, true);
+					}
+					else {
+						$val .= $portion;
+					}
+				}
+				self::setVariableByAlias($config, $val);
+			}
+		}
+		else {
+			$val = "";
+			if(is_array($config)) {
+				if(isset($config["check"])) {
+					$matches = array();
+					if(isset($config["check"][1])) {
+						if(preg_match("/{(.*)}/si", $config["check"][1], $matches)) {
+							$val1 = self::getVariableByAlias($matches[1]);
+						}
+						else {
+							$val1 = $config["check"][1];
+						}
+					}
+
+					if(isset($config["check"][2])) {
+						if(preg_match("/{(.*)}/si", $config["check"][2], $matches)) {
+							$val2 = self::getVariableByAlias($matches[2]);
+						}
+						else {
+							$val2 = $config["check"][2];
+						}
+					}
+
+					switch($config["check"][0])
+					{
+						case "=":
+							if($val1 == $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case "<>":
+							if($val1 != $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case "<":
+							if($val1 < $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case ">":
+							if($val1 > $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case "<=":
+							if($val1 <= $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+						case ">=":
+							if($val1 >= $val2) {
+								return $this->parseDynamicConfigs($config["true"]);
+							}
+							else {
+								return $this->parseDynamicConfigs($config["false"]);
+							}
+							break;
+					}
+				}
+				else {
+					foreach($config as $portion) {
+						if(is_array($portion)) {
+							$val .= $this->parseDynamicConfigs($portion, true);
+						}
+						else {
+							if($resolve) {
+								$portionVal = self::getVariableByAlias($portion);
+								if(is_array($portionVal)) {
+									$portionVal = $this->parseDynamicConfigs($portionVal);
+								}
+								$val .= $portionVal;
+							} else {
+								$val .= $portion;
+							}
+						}
+					}
+					return $val;
+				}
+			}
+			else {
+				return $this->parseDynamicConfigs(self::getVariableByAlias($config));
```

Comment:
I think that this was supposed to be reverted, it was added in an attempt to fix the race condition for conditional configurations.